### PR TITLE
Fix issues 7, 8, 9, 10 and readme update

### LINKED ISSUE #7: typo - bq.go
typo, "Bigqueyr": https://github.com/GoogleCloudPlatform/DIY-Tools/blob/21f62b69ce1ae14df1d5d17fd9cb06bec657814a/gcp-data-drive/bq.go#L40

### LINKED ISSUE #8: Comments Typo
https://github.com/GoogleCloudPlatform/DIY-Tools/blob/21f62b69ce1ae14df1d5d17fd9cb06bec657814a/gcp-data-drive/bq.go#L78

### LINKED ISSUE #9: Fix Functionality descriptions
"Getting data from BigQuery", "In the following snippet, the for statement collects the results into a map named res." res is a slice of maps, not a single map. Similar confusion in the actual code at https://github.com/GoogleCloudPlatform/DIY-Tools/blob/21f62b69ce1ae14df1d5d17fd9cb06bec657814a/gcp-data-drive/bq.go#L49 and https://github.com/GoogleCloudPlatform/DIY-Tools/blob/21f62b69ce1ae14df1d5d17fd9cb06bec657814a/gcp-data-drive/fs.go#L61.

### LINKED ISSUE #10: Build parameters for app.
"go build -mod=readonly" -- -mod=readonly is unnecessary here.

### LINKED ISSUE #11: Remove inconsistent newline after function signatures.

### LINKED ISSUE #12: Update the deploy readme with the new deployment types.
Update the deploy readme with new deployment types

### PR DESCRIPTION
**Pull request by goog-lukemc**
This PR is various comment correction in the source code based on feedback from the Go engineering team. Fixes #7 Fixes #8 Fixes #9  Fixes #10 Fixes #11 Fixes #12 . This PR also includes README.md updates from Fixes #13 .

### CODE REVIEW

File: gcp-data-drive/bq.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (b *bqDataPlatform) close() error {
 	return nil
 }
 
-// getBQInterface creates and populates the Bigquery platform client requirementa and returns
+// getBQInterface creates and populates the Bigquery platform client requirements and returns
 // a usable getData() method.
```
Comment (tbpg) [inline gcp-data-drive/bq.go:79]: Might need to change more of this. Perhaps

```suggestion
// getBQInterface creates a BigQuery client and returns a TODO that gets data from BigQuery.
```

I don't remember the name of the interface `*bqDataPlatform` implements -- that's what I mean by the TODO. This doesn't return a `getData` method directly.

File: gcp-data-drive/fs.go [hunk 2]
```diff
@@ -58,7 +57,7 @@ func (f *fsDataPlatform) getData(ctx context.Context) ([]byte, error) {
 		return nil, err
 	}
 
-	// Create a map to hold our firestore result set.
+	// Create a slice of map string interface to hold our firestore result set.
```
Comment (tbpg) [inline gcp-data-drive/fs.go:60]:
```suggestion
	// Create a slice to hold our Firestore results.
```

File: gcp-data-drive/README.md [hunk 3]
```diff
@@ -2,16 +2,36 @@
...
-:warning: Edit the deploy script and app.yaml for your needs. Used as is, the command below will deploy and new version of the default App Engine service.
+### Deploy to multiple compute platforms  
+Google Cloud SDK must be installed and initialized to preform these actions. Edit the PROJECT_ID variable as required.
 
+#### Cloud Run
 ```bash
-git clone https://github.com/GoogleCloudPlatform/DIY-Tools.git
-cd DIY-Tools/gcp-data-drive
-./appengine_deploy.sh {gcp_projectid}
+PROJECT_ID=YourProjectID
```
[the remark this answers is not in the document]
Comment (goog-lukemc) [reply gcp-data-drive/README.md:12]: I agree. Unfortunately, these variable names are used in the qwiklab. I'll create a separate issue to sync these variable names the next time we revise the qwiklab content.

File: gcp-data-drive/README.md [hunk 4]
```diff
@@ -2,16 +2,36 @@
 GCP Data Drive leverages a composable url path to retrieve data in JSON formats from supported GCP data platforms. Bigquery and Firestore are currently supported.
 
 ## Installation
-### Deploy to App Engine
-The shell script below can be executed from a bash shell.
+Clone this repo. Change directory to gcp-data-drive.
```
[the remark this answers is not in the document]
Comment (goog-lukemc) [reply gcp-data-drive/README.md:5]: Added per suggestion

File: gcp-data-drive/README.md [hunk 5]
```diff
@@ -2,16 +2,36 @@
...
 
-:warning: Edit the deploy script and app.yaml for your needs. Used as is, the command below will deploy and new version of the default App Engine service.
+### Deploy to multiple compute platforms  
+Google Cloud SDK must be installed and initialized to preform these actions. Edit the PROJECT_ID variable as required.
```
[the remark this answers is not in the document]
Comment (goog-lukemc) [reply gcp-data-drive/README.md]: Added per suggestion

File: gcp-data-drive/fs.go (hunk 2 above)
Comment (goog-lukemc) [reply gcp-data-drive/fs.go]: Updated to "slice of maps" to be consistent with the BQ comments.

File: gcp-data-drive/bq.go (hunk 1 above)
Comment (goog-lukemc) [reply gcp-data-drive/bq.go]: I corrected with some wording I think is better.

File: gcp-data-drive/bq.go [hunk 6]
```diff
@@ -37,7 +37,7 @@ type bqDataPlatform struct {
 	query *bigquery.Query
 }
 
-// getData contains the implementation detail for retriving and marshaling data from Bigqueyr into JSON.
+// getData contains the implementation detail for retriving and marshaling data from Bigquery into JSON.
```
Comment (tbpg) [inline gcp-data-drive/bq.go:40]: Whoops, `BigQuery` (the Q is capitalized).

Comment (goog-lukemc) [reply gcp-data-drive/bq.go:40]: I correct all occurrences in this file.

**Comment by tbpg** [comment]
Need a push?

**Pull request merged**